### PR TITLE
Codegen for access helpers to Tuple data structures

### PR DESCRIPTION
**Pull request by fordN**
Part of the Solidity structs epic: [graph-node issue #766](https://github.com/graphprotocol/graph-node/issues/766)

This PR updates `graph codegen` to support `Tuple` access in the mappings.  It now generates:
  - A class per Tuple with accessors for each of it's parameters. 
  - Tuple type accessors on the event that convert to a Tuple and reference one of the generated Tuple classes.

### CODE REVIEW

**Comment by Jannis** [comment]
How is this different from #235?

File: src/codegen/abi.js [hunk 1]
```diff
@@ -54,28 +56,15 @@ module.exports = class AbiCodeGenerator {
...
-            ),
+          // Generate getters and classes for event params
+          let paramObject = this._generateEventParam(
+            input,
+            index,
+            eventClassName,
+            `event`,
           )
+          paramsClass.addMethod(paramObject.getter)
+          tupleClasses.push(...paramObject.classes)
```
Comment (Jannis) [inline src/codegen/abi.js]: Hehe, I guess this could also just be `tupleClasses.concat(paramObject.classes)`.

Comment (fordN) [reply src/codegen/abi.js]: This allows us to change it in place though and not create a new reference :).

Comment (Jannis) [reply src/codegen/abi.js]: Ah, good point!

**Pull request merged**